components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationsApi.java

```diff
@@ -804,11 +823,37 @@ public Response importApplication(@Multipart(value = "file", required = false) I
         @ApiResponse(code = 409, message = "Conflict", response = Error.class),
         @ApiResponse(code = 500, message = "Server Error", response = Error.class)
     })
-    public Response importApplicationForUpdate(@Multipart(value = "file", required = false) InputStream fileInputStream,@Multipart(value = "file" , required = false) Attachment fileDetail) {
+    public Response importApplicationForUpdate(@Multipart(value = "file", required = false) InputStream fileInputStream, @Multipart(value = "file" , required = false) Attachment fileDetail) {
 
         return delegate.importApplicationForUpdate(fileInputStream, fileDetail );
     }
 
+    @Valid
+    @POST
+    @Path("/import/{fileType}")
+    @Consumes({ "multipart/form-data" })
+    @Produces({ "application/json" })
+    @ApiOperation(value = "Create application from an exported file ", notes = "This API provides the capability to store the application information, provided as a file. ", response = Void.class, authorizations = {
+            @Authorization(value = "BasicAuth"),
+            @Authorization(value = "OAuth2", scopes = {
+            })
+    }, tags={ "Applications", })
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Successfully created.", response = Void.class),
+            @ApiResponse(code = 400, message = "Bad Request", response = Error.class),
+            @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+            @ApiResponse(code = 409, message = "Conflict", response = Error.class),
+            @ApiResponse(code = 500, message = "Server Error", response = Error.class)
+    })
+    public Response importApplicationFromFile(
+            //todo
```

Comment:
Can you mention what is needed to be done ?

Reply:
Refactored --> todo : get file type from fileDetail or Content-Type header instead of path parameter

Sorry I should have specified.